File: P120882.py
def solution(score):
    answer = [0] * len(score)
    score_ = sorted(score, reverse=True, key=lambda x:x[0]+x[1])
    rank = {}
    cnt = 1
    for sc in score_:
        if rank.get(sum(sc)):
            cnt += 1
        else:
            rank[sum(sc)] = cnt
            cnt += 1
    for i in range(len(score)):
        answer[i] = rank[sum(score[i])]
    
    return answer

File: P120882.java
import java.util.Arrays;
import java.util.HashMap;

public class P120882 {
        public int[] solution(int[][] score) {
        int[] answer = new int[score.length];
        int[][] scores = new int[score.length][score[0].length];
        for (int i=0; i<score.length; i++) {
            System.arraycopy(score[i], 0, scores[i], 0, score[0].length);
        }
        
        Arrays.sort(scores, (x, y) -> y[0]+y[1]-x[0]-x[1]);
        HashMap<Integer, Integer> rank = new HashMap<>();
        int cnt = 1;
        for (int[] sc : scores) {
            if (rank.containsKey(sc[0]+sc[1]))cnt += 1;
            else {
                rank.put(sc[0]+sc[1], cnt);
                cnt += 1;
            }
        }
        for (int i=0; i<score.length; i++) answer[i] = rank.get(score[i][0]+score[i][1]);
        return answer;
    }
}
